stdlib/builtins.pyi

```diff
@@ -1561,15 +1562,15 @@ else:
     def sum(__iterable: Iterable[bool], __start: int = ...) -> int: ...  # type: ignore[misc]
 
 @overload
-def sum(__iterable: Iterable[_SumT]) -> _SumT | Literal[0]: ...
+def sum(__iterable: Iterable[_AddableWithIntT]) -> _AddableWithIntT | Literal[0]: ...
```

Comment:
This doesn't seem right:

```python
In [2]: class Foo:
   ...:     def __add__(self, other):  # accepts int as "other"
   ...:         return Foo()
   ...: 

In [3]: sum([Foo(), Foo()])
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-3-55dd5d052a18> in <module>
----> 1 sum([Foo(), Foo()])

TypeError: unsupported operand type(s) for +: 'int' and 'Foo'
```

Reply:
That's a weird `TypeError` there though. Good spot. Maybe `__radd__` is involved somehow.

Reply:
Indeed, `__add__` and `__radd__` are both required if no default is specified:

```python
>>> class Foo:
...     def __add__(self, other):
...         return Foo()
...
>>> sum([Foo(), Foo()])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for +: 'int' and 'Foo'
>>> class Bar:
...     def __radd__(self, other):
...         return Bar()
...
>>> sum([Bar(), Bar()])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for +: 'Bar' and 'Bar'
>>> class Baz:
...     def __add__(self, other):
...         return Baz()
...     def __radd__(self, other):
...         return Baz()
...
>>> sum([Baz(), Baz()])
<__main__.Baz object at 0x000001E71D90BD30>
```